nemo/collections/asr/models/label_models.py

```diff
@@ -87,39 +86,62 @@ def list_available_models(cls) -> List[PretrainedModelInfo]:
 
     def __init__(self, cfg: DictConfig, trainer: Trainer = None):
         self.world_size = 1
-        self.cal_labels_occurrence = False
+        self.cal_labels_occurrence_train = False
+        self.labels_occurrence = None
+
+        if 'loss' in cfg:
+            if 'weight' in cfg.loss:
+                if cfg.loss.weight == 'auto':
+                    self.cal_labels_occurrence_train = True
+                else:
+                    weight = cfg.loss.weight
+            else:
+                weight = None  # weight is None for angular loss and CE loss if it's not specified.
+
         if trainer is not None:
             self.world_size = trainer.num_nodes * trainer.num_devices
 
         super().__init__(cfg=cfg, trainer=trainer)
 
-        self.preprocessor = EncDecSpeakerLabelModel.from_config_dict(cfg.preprocessor)
-        self.encoder = EncDecSpeakerLabelModel.from_config_dict(cfg.encoder)
-        self.decoder = EncDecSpeakerLabelModel.from_config_dict(cfg.decoder)
+        if self.labels_occurrence:
+            # Goal is to give more weight to the classes with less samples so as to match the ones with the higher frequencies
```

Comment:
do we need this comment?

Reply:
let's keep it. This is to let user to understand how we calculate weight easily